### PR TITLE
Mono compatibility

### PR DESCRIPTION
**Pull request by s-tarasov**

### CODE REVIEW

**Comment by ferventcoder** [comment]
I'd almost prefer WindowsFileSystemAccess be renamed - see the evolution of it here https://github.com/chocolatey/choco/blob/master/src/chocolatey/infrastructure/filesystem/DotNetFileSystem.cs

**Comment by s-tarasov** [comment]
DotNet cross platform is coming)

File: product/roundhouse/infrastructure.app/logging/log4net.config.xml [hunk 1]
```diff
@@ -1,23 +1,23 @@
 ﻿<?xml version="1.0"?>
 <log4net>
-  <appender name="ColoredConsoleAppender" type="log4net.Appender.ColoredConsoleAppender">
+  <appender name="ColoredConsoleAppender" type="log4net.Appender.ManagedColoredConsoleAppender">
```
Comment (ferventcoder) [inline product/roundhouse/infrastructure.app/logging/log4net.config.xml]: :+1:

File: product/roundhouse/infrastructure/filesystem/DotNetFileSystemAccess.cs [hunk 2]
```diff
@@ -13,8 +13,10 @@ namespace roundhouse.infrastructure.filesystem
     /// <summary>
     /// All file system access code comes through here
     /// </summary>
-    public sealed class WindowsFileSystemAccess : FileSystemAccess
+    public sealed class DotNetFileSystemAccess : FileSystemAccess
     {
+        private static readonly bool is_running_on_mono = Type.GetType("Mono.Runtime") != null;
```
Comment (ferventcoder) [inline product/roundhouse/infrastructure/filesystem/DotNetFileSystemAccess.cs]: That's all it takes? :)

**Comment by ferventcoder** [comment]
Still don't believe the log4net upgrade is required but perhaps @ChrisMissal can look into that.

File: product/roundhouse/infrastructure/filesystem/DotNetFileSystemAccess.cs (hunk 2 above)
Comment (s-tarasov) [inline product/roundhouse/infrastructure/filesystem/DotNetFileSystemAccess.cs]: Also need add DbProviderFactories section in app.config
``` xml
 <system.data>
        <DbProviderFactories>
          <remove invariant="MySql.Data.MySqlClient" />
          <add name="MySQL Data Provider" invariant="MySql.Data.MySqlClient" description=".Net Framework Data Provider for MySQL" type="MySql.Data.MySqlClient.MySqlClientFactory, rh" />
        </DbProviderFactories>
    </system.data>
 ```

I'm not testing all features. Only simplest rh configuration and mysql provider..

**Comment by drusellers** [comment]
@ferventcoder yes, the upgrade is needed. Its not in the version RH has currently.

File: product/roundhouse/infrastructure/filesystem/DotNetFileSystemAccess.cs (hunk 2 above)
Comment (drusellers) [inline product/roundhouse/infrastructure/filesystem/DotNetFileSystemAccess.cs]: > That's all it takes? :)

That's all TopShelf does as well.

**Comment by drusellers** [comment]
@ChrisMissal @ferventcoder I'd love to get this merged in. this is the real blocker to mono adoption.

**Comment by drusellers** [comment]
YAS :clap:

**Pull request merged**